src/themes/color.rs

```diff
@@ -216,6 +216,21 @@ impl FromStr for Color {
             let v = components.next().or_error(err_msg)??;
             let a = components.next().unwrap_or(Ok(100.))?;
             Color::Hsva(Hsva::new(h, s / 100., v / 100., (a / 100. * 255.) as u8))
+        } else if color.starts_with("x:") {
+            let name = color.split_at(2).1;
+            let hex = super::xresources::get_color(name)?
+                .ok_or_else(|| {
+                    Self::Err::new(format!("color {} not defined in ~/.Xresources", name))
+                })?;
+            let err_msg = || format!("'{name}' def '{hex}' cannot be parsed as RGB");
+            let rgb = hex
+                .get(1..7)
+                .map(|rgb| u32::from_str_radix(rgb, 16).ok())
+                .flatten();
+            let a = u32::from_str_radix(hex.get(7..9).unwrap_or("FF"), 16).ok();
+            rgb.map(|rgb| a.map(|a| Color::Rgba(Rgba::from_hex((rgb << 8) + a))))
+                .flatten()
+                .or_error(err_msg)?
```

Comment:
Can't we just return `hex.parse()`?

Reply:
Or `hex.parse().or_error(|| format!("{name} definition is invalid"))?`

Reply:
But `hex.parse()` would work with alpha too?

Reply:
Yea, but not without. 😏

Reply:
I don't get it. Could you show an example of .Xresources for which it wouldn't work?

Reply:
`#ffeeddaa` parses as R: 255, G: 238, B: 221, A: 170.
`#ffeedd` parses as R:0, G: 255, B:238, A: 221.

Reply:
Wait, your regex strips the alpha.

Reply:
Oh, true.So, I could add an alpha group and default it to ff. Then `hex.parse()` would work, right?

Reply:
It already works. I don't know how useful alpha channel in .Xresources is, so IDK.

Reply:
Have a look at the latest version, I think this is better.